src/providers/wms/qgswmsdataitems.cpp

```diff
@@ -115,7 +115,18 @@ QVector<QgsDataItem *> QgsWMSConnectionItem::createChildren()
     for ( const QgsWmtsTileLayer &l : constTileLayers )
     {
       QString title = l.title.isEmpty() ? l.identifier : l.title;
-      QgsDataItem *layerItem = l.styles.size() == 1 ? this : new QgsDataCollectionItem( this, title, mPath + '/' + l.identifier );
+      QgsDataItem *layerItem;
+
+      // Using if condition, ternary operator leads to type casting issue
```

Comment:
you can avoid this by inserting explicit casts:

    QgsDataItem *layerItem = l.styles.size() == 1 ? static_cast<  QgsDataItem * >( this ) : static_cast<  QgsDataItem *>( new QgsDataCollectionItem( this, title, mPath + '/' + l.identifier ); )

Reply:
Got it!